x/observer/keeper/msg_server_disable_fast_confirmation.go

```diff
@@ -0,0 +1,56 @@
+package keeper
+
+import (
+	"context"
+
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	authoritytypes "github.com/zeta-chain/node/x/authority/types"
+	"github.com/zeta-chain/node/x/observer/types"
+)
+
+// DisableFastConfirmation disables fast confirmation for the given chain ID
+// Inbound and outbound will be only confirmed using SAFE confirmation count on disabled chains
+func (k msgServer) DisableFastConfirmation(
+	goCtx context.Context,
+	msg *types.MsgDisableFastConfirmation,
+) (*types.MsgDisableFastConfirmationResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// check permission
+	err := k.GetAuthorityKeeper().CheckAuthorization(ctx, msg)
+	if err != nil {
+		return nil, errorsmod.Wrap(authoritytypes.ErrUnauthorized, err.Error())
+	}
+
+	// find current chain params list
+	chainParamsList, found := k.GetChainParamsList(ctx)
+	if !found {
+		return &types.MsgDisableFastConfirmationResponse{}, types.ErrChainParamsNotFound
```

Comment:
Inconsistent, sometime we return nil and sometimes an empty value
Let's return nil everywhere

Reply:
updated in the same commit https://github.com/zeta-chain/node/pull/3522/commits/05ece1e475ab004869bd8297f35127d5d9870f31